MNTOR-3924: Plus Expiration heads-up page

Vinnl opened the pull request:
<!-- The following is intended to be helpful to you. Feel free to remove anything that is not. -->

# References:

Jira: MNTOR-3924
Figma:

<!-- When adding a new feature: -->

# Description

Note: I did not add unit tests yet because it looks like Jest does not currently have access to the same version of React as our app does, and hence, useFormState is not supported:
https://stackoverflow.com/a/78736908

Once we upgrade to React 19 (and use useActionState instead), the tests can be added.

The same is also a requirement for showing the pending state, which requires the useActionState API.

It's also not finished yet because it's waiting on the back-end code to know which subscriptions are about to expire: https://github.com/mozilla/blurts-server/pull/5512

However, since it's all under pretty tight timelines, I figured I'd submit this already to get a design and first code review, and finish it up once the other pieces are in place.

# Screenshot (if applicable)

See [Storybook](https://deploy-preview-5524--fx-monitor-storybook.netlify.app/?path=/story/pages-logged-in-plus-expiration), also for the [terms](https://deploy-preview-5524--fx-monitor-storybook.netlify.app/?path=/story/pages-public-terms-plus-expiration-offer--expiration-offer-terms).

# How to test

I'm not sure if it's possible to test the coupon flow locally and if it's entirely done yet, but you should be able to view the individual pages in Storybook at least, and see the happy path and terms on the dev server.

# Checklist (Definition of Done)

- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug. - No, see above
- [x] If this PR implements a feature flag or experimentation, I've checked that it still works with the flag both on, and with the flag off.
- [x] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

flodolo commented on locales-pending/plus-expiration.ftl:
```diff
@@ -0,0 +1,37 @@
+plus-expiration-error-already-applied-content-part2 = <support-link>Contact our Support team</support-link> for help or head to your dashboard to see the latest on how we’re protecting your data.
+plus-expiration-error-already-applied-cta-label = Go to { -brand-monitor }
+
+plus-expiration-terms-heading = <b>Terms and restrictions</b> 10% off subscription renewal
```
Is it just me or this string seems a bit strange? How does it look in the website?

Vinnl replied:
Heh, now that you mention it it does, but it looked fine initially, and I think it's technically correct (though maybe needs a dash somewhere) - it's the terms for getting 10% off of the renewal of your subscription.

Here's what it looks like:

![image](https://github.com/user-attachments/assets/60dd870c-04d2-4a38-afe6-20ed42da2800)

Vinnl replied:
I'll check with content.

Vinnl replied:
OK, they're on separate lines now, and therefore, also separate strings:

![image](https://github.com/user-attachments/assets/0b1a67e6-f555-4fe0-a094-bd5c08082b54)

Vinnl commented on .storybook/preview.tsx:
```diff
@@ -74,6 +74,10 @@ const preview: Preview = {
             linkTo("Pages/Public/Breach listing")();
           }
 
+          if (path === "/terms/expiration-offer") {
```
It doesn't look like this actually works anymore with the latest Storybook (I can't find the docs on it anymore either), but figured it couldn't harm in case they add it back or something.

flozia commented on src/app/(proper_react)/(redesign)/terms/expiration-offer/View.tsx:
The line height of the `h1` seems a bit too high as well as the indentation of the list. On mobile the content overlaps with the header logo.

<img width="332" alt="image" src="https://github.com/user-attachments/assets/d12b1ef5-61b4-492f-9e22-e37b38bcd82f" />

Vinnl replied:
Good callout, fixed in 66c32df8e, and a44d106dd04c0d33be7611773d28b10320af5154 applied the title font token to fix the line height.

flozia commented on src/app/(proper_react)/(redesign)/terms/expiration-offer/View.tsx:
```diff
@@ -0,0 +1,47 @@
+          <li>{l10n.getString("plus-expiration-terms-term5")}</li>
+          <li>
+            {l10n.getFragment("plus-expiration-terms-term6", {
+              elems: { "monitor-link": <Link href="/" /> },
```
If we don’t know the `href` value, yet: Should we add a TODO comment?

Vinnl replied:
We do know the value; it's a link to the homepage (`/`) :)

![image](https://github.com/user-attachments/assets/e05fd7c7-7410-4bc2-929b-3d906356ace3)

Vinnl commented:
Rebased and added 4d6d463d857bfed4fc259ab781b0c24850fa8b12 to use `useActionState` now that we're on React 19, and 44357727c5baf149e90cf83b79f50ba1cc7ba534 to add `@storybook/test` to use the `fn` function for actions. (Technically a new dependency, but since it's part of Storybook and gets updated with it, figured I could just add it.)
